frontend/src/views/consumer/event-listing/EventBooking.jsx

```diff
@@ -0,0 +1,283 @@
+    }
+    & form {
+      & > * {
+        margin-top: 0.5rem;
+      }
+    }
+  }
+`;
+const CustomerGroupButton = styled(Button)`
+  &&& {
+    padding: 0.5rem;
+    width: 100%;
+    height: auto;
+    font-size: 0.86rem;
+    border-radius: 8px;
+  }
+`;
+const FlexBoxHorizontal = styled(Collapsible)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 0.5rem 0;
+
+  & > *:first-child {
+    margin-right: 1rem;
+  }
+`;
+const TotalAmountCalculated = styled(FlexBoxHorizontal)`
+  background-color: rgba(0, 0, 0, 0.1);
+  padding: 0.5rem 1rem;
+  border-radius: 8px;
+  justify-content: space-between;
+  overflow: hidden;
+  * {
+    margin-bottom: 0;
+  }
+`;
+
+// child components
+const FormListItem = props => {
+  return (
+    <FormListItemWrapper>
+      <Typography type="largebody">{props.title}</Typography>
+      <Collapsible pose={props.disabled ? 'collapsed' : 'natural'}>
+        {props.children}
+      </Collapsible>
+    </FormListItemWrapper>
+  );
+};
+
+// main class
+export default connect('store')(
+  class EventBooking extends Component {
+    internalState = observable({
+      customerGroup: undefined,
+      tickets: [
+        {
+          id: 'first',
+          label: this.props.event.ticketCatalog[0].ticketDescription,
+          value: this.props.event.ticketCatalog[0].id,
+          amount: 0,
+        },
+      ],
+      phoneNumber: '',
+      name: '',
+      email: '',
+      classRoom: '',
+      school: '',
+    });
+    getTotalPrice = () => {
+      return this.internalState.tickets.reduce((acc, curr) => {
+        const targetCatalog = this.props.event.ticketCatalog.find(
+          elem => elem.id == curr.value,
+        );
+        if (targetCatalog) {
+          const singlePrice = targetCatalog.price;
+          const totalPriceForThisCatalog = singlePrice * curr.amount;
+          acc += totalPriceForThisCatalog;
+        }
+        return acc;
+      }, 0);
+    };
+    render() {
+      const { event } = this.props;
+      const { internalState } = this;
+
+      const isPrivateCustomer = internalState.customerGroup == 'private';
+      const isGroupConductorCustomer = internalState.customerGroup == 'group';
+      const totalCost = this.getTotalPrice();
+      const submittable = isGroupConductorCustomer
+        ? internalState.school &&
+          internalState.classRoom &&
+          isValidNumber(internalState.phoneNumber, 'FI')
+        : internalState.name && isValidNumber(internalState.phoneNumber);
+      return (
+        <Wrapper bgColor={event.themeColor}>
+          <Typography type="title">Varaa/osta lippu</Typography>
+          <FormListItem title="1 - Olen">
+            <ButtonGroup style={{ display: 'flex' }}>
+              <CustomerGroupButton
+                backgroundColor={
+                  isGroupConductorCustomer ? event.themeColor : 'white'
+                }
+                onClick={e => (internalState.customerGroup = 'group')}
```

Comment:
Inlining these actions here makes this component hard to test.. even though the action is currently simple and just a one-liner, I would prefer making it a method of the component instead.

Reply:
I agree with you about the test. One problem is that we don't even have the resources to test these components (We dont even test front-end state management codes!).
I actually refactored methods into inline like this. Makes it more visible.